ntp-proto/src/packet.rs

```diff
@@ -113,25 +114,162 @@ pub struct NtpPacket<'a> {
     mac: Option<Mac<'a>>,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Clone, PartialEq, Eq)]
 pub enum ExtensionField<'a> {
-    Unknown { typeid: u16, data: Cow<'a, [u8]> },
+    UniqueIdentifier(Cow<'a, [u8]>),
+    NtsCookie(Cow<'a, [u8]>),
+    NtsCookiePlaceholder {
+        body_length: u16,
+    },
+    NtsEncryptedField {
+        nonce: Cow<'a, [u8]>,
```

Comment:
The nonce is essentially part of the details of encryption/decryption, and shouldn't be needed anywhere else, so I would suggest we just not keep it around here.